CIPs/0017.md

```diff
@@ -0,0 +1,69 @@
+
+# CIP [0017]: Integrate Plumo Proofs into the lightest sync protocol
+
+- Date: 2020-07-27
+- Author: @lucasege
+- Status: WIP
+
+## Overview
+
+In order to continue supporting the dream of a seamless mobile sync experience, the lightest sync protocol must be extended to support PLUMO Zk-SNARKs which can quickly verify transitions from epoch X to epoch X + Y, where Y could be on the order of 10s-100s of epochs.
+
+The prover/verifier for plumo proofs have already been implemented, and the exposed go interface is available [here](https://github.com/celo-org/celo-bls-go)
+
+## Goals
+
+- Support adding/storing proofs on full nodes
+- Gossip these proofs to propogate them through the network
+- Extend the `les` protocol such that the server will provide these proofs in place of headers (when appropriate), and the client will consume/verify proofs accordingly.
+
+## Proposed Solution
+
+We propose to add two new RPCs: 
+`AddProof(proof []byte, proofId uint)`
```

Comment:
Will this RPC call verify that the plumo proof is valid?  Or does that even need to be done? (I don't know enough about the plumo/SNARK proofs to know if that's even a valid question).

Reply:
Yes in my mind this could and should be done. This would help prevent potential for DDoS via someone spamming bad proofs to one node which then propagate to the whole network. I'll add this to the CIP after I finish synthesizing the rest of the feedback